corpus/filter.py

```diff
@@ -242,7 +242,7 @@ def tasks(self):
     def run(self):
         segments = []
         for seg in self.segment_file_list:
-            with open(tk.uncached_path(seg)) as f:
+            with uopen(tk.uncached_path(seg)) as f:
```

Comment:
There is a difference: `open` will always try to open in text mode. `uopen` is somewhat unspecific for that without options, so it could be that you get text mode (`readlines()` returns `list[str]`) or bytes (`readlines()` returns `list[bytes]`). I don't think you want to allow for both. So maybe just do:
```suggestion
            with uopen(tk.uncached_path(seg), "rt") as f:
```